base/intfuncs.jl

```diff
@@ -477,12 +477,12 @@ julia> nextpow(4, 16)
 16
 ```
 """
-function nextpow(a::Real, x::Real)
+function nextpow(a::T, x::Real) where T <: Real
     x <= 0 && throw(DomainError(x, "`x` must be positive."))
     # Special case fast path for x::Integer, a == 2.
     # This is a very common case. Constant prop will make sure that a call site
     # specified as `nextpow(2, x)` will get this special case inlined.
-    a == 2 && isa(x, Integer) && return _nextpow2(x)
+    a == 2 && isa(x, Integer) && return T(_nextpow2(x))
```

Comment:
Is there an implicit contract that `x isa Integer` must satisfy `typeof(x * x) == typeof(x)`? Long ago that wasn't true, now it's true for Base types, but is it actually required?